python/ray/data/datasource/datasink.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Iterable, List
+
+from ray.data._internal.execution.interfaces import TaskContext
+from ray.data.block import Block
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class Datasink:
+    def on_write_start(self, **_) -> None:
+        """Callback for when a write job starts.
+
+        Use this method to perform setup for write tasks. For example, creating a
+        staging bucket in S3.
+
+        Args:
+            _: Forward-compatibility placeholder.
```

Comment:
Are we allowing passing arbitrary arguments here? Did you see the same pattern from other code base as well? I am worried about the thing get abused, and hard to manage later. Can't we add more arguments in the future when we need?

Reply:
Yeah, we used forward compatability placeholders before. Advantage of forward compability place holder is that we can add parameters without breaking user's subclasses. Disadvantage is that they can get misused.

Given that this is a developer API, I think it's okay if we add parameters between releases, and I've removed the forward compability placeholders.